### PR TITLE
obsidian: init at 0.7.3

### PR DESCRIPTION
**Pull request by ConradMearns**
###### Motivation for this change

Adds Obsidian editor version 0.7.3 (#89089)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/obsidian/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  profile = ''
+    export LC_ALL=C.UTF-8
+    export XDG_DATA_DIRS=${gsettings-desktop-schemas}/share/gsettings-schemas/${gsettings-desktop-schemas.name}:${gtk3}/share/gsettings-schemas/${gtk3.name}:$XDG_DATA_DIRS
+  '';
+
+  meta = with lib; {
+    description = "Obsidian is a powerful knowledge base that works on top of a local folder of plain text Markdown files.";
+    homepage = "https://obsidian.md";
+    license = licenses.mit;
```
Comment (iblech) [inline pkgs/applications/misc/obsidian/default.nix:25]: Thank you for packaging Obsidian! :-) I think you mean `licenses.obsidian` here, right?

Comment (ConradMearns) [reply pkgs/applications/misc/obsidian/default.nix]: :sweat_smile:  good catch! Thanks:)

File: pkgs/applications/misc/obsidian/default.nix [hunk 2]
```diff
@@ -19,6 +19,9 @@ appimageTools.wrapType2 rec {
     export XDG_DATA_DIRS=${gsettings-desktop-schemas}/share/gsettings-schemas/${gsettings-desktop-schemas.name}:${gtk3}/share/gsettings-schemas/${gtk3.name}:$XDG_DATA_DIRS
   '';
 
+  # Strip version from binary name.
+  extraInstallCommands = "mv $out/bin/${name} $out/bin/${pname}";
```
Comment (ConradMearns) [inline pkgs/applications/misc/obsidian/default.nix:23]: I get mixed results with this... When running `nix-build -A obsidian` from the root of my cloned nixpkgs folder, the operation works correctly. However, in my /etc/nixos/config, if I import my custom branch and include obsidian, then it remains `obsidian-0.7.3`

Comment (ConradMearns) [reply pkgs/applications/misc/obsidian/default.nix]: Found a good example in the notable package. Tested the fix on a seperate branch, got the expected name.

**Pull request merged**